blog/2023-01-17-mongodb-monitoring.md

```diff
@@ -0,0 +1,297 @@
+ 
+![image](https://user-images.githubusercontent.com/24860547/212876934-c79a8581-366a-4d16-8782-7f6e94a3ac54.png)
+
+ - **WiredTiger Lock Duration**: The duration of locks held in the WiredTiger storage engine on the MongoDB server. 
+
+If the lock duration is too long, it can indicate that the database may be trying to access or modify data that is locked by another thread, leading to a deadlock. This can result in degraded query performance, or even data corruption.
+
+One way to prevent this issue is to regularly monitor the lock duration on the server. If the lock duration is too long, it can indicate an issue that needs to be addressed. Additionally, increasing the WiredTiger cache size can help reduce the lock duration by allowing more data to be stored in memory, which can reduce the amount of disk I/O required.
+
+ - **WiredTiger Log Operations**: The number of operations written to the WiredTiger log on the MongoDB server. 
+ 
+Keeping an eye on the number of operations written to the WiredTiger log is important as it can indicate how much disk space is being used, and if it is increasing excessively it can suggest that there is an issue with disk space or with the database configuration. For example, if the log operations are rapidly increasing it could suggest that the database is not being properly indexed or that queries are not being optimised, resulting in an increase in disk usage.
+ 
+ ![image](https://user-images.githubusercontent.com/24860547/212877037-64b89f38-3323-4daf-b267-f38d4414dbc7.png)
+
+  - **WiredTiger Log Operations Size**: The size of operations written to the WiredTiger log on the MongoDB server. 
+  
+![image](https://user-images.githubusercontent.com/24860547/212877086-61fffbf7-2254-45c8-bb07-8ad5bf2601d0.png)
+
+ - **WiredTiger Transactions**:The number of transactions currently open in the WiredTiger storage engine on the MongoDB server. 
+ 
+Issues that can be prevented through effective monitoring of this metric include identifying slow queries, detecting potential deadlocks, and identifying any transactions that are taking longer than expected to commit.
+
+![image](https://user-images.githubusercontent.com/24860547/212877136-bc34c31f-827a-4366-9a5c-8907b9a5de82.png)
+
+### Database 
+ - **Database Collections**: The number of collections in the specified database on the MongoDB server. 
```

Comment:
Chart snapshot missing.. Good to document if specific configuration is needed on MongoDB? Comment valid for this entire section and subsequent sections on replication, shards etc.